.github/workflows/tests.yaml

```diff
@@ -0,0 +1,83 @@
+name: tests
+on:
+  pull_request:
+    paths:
+    - 'src/**'
+    - '.github/workflows/tests.yaml'
+    - '!src/rez/utils/_version.py'
+    - '!**.md'
+  push:
+    paths:
+    - 'src/**'
+    - '.github/workflows/tests.yaml'
+    - '!src/rez/utils/_version.py'
+    - '!**.md'
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  core:
+    name: Core tests
+
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        os: ['macos-11', 'ubuntu-latest', 'windows-2019']
```

Comment:
Note that I'm still using macOS 11 here. Tests are failing on the latest because python can't be found from within the packages (generated/built during the tests). We will have to figure this out before merging.